packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -123,11 +124,52 @@ class CupertinoCheckbox extends StatefulWidget {
 
   /// The color to use when this checkbox is checked.
   ///
+  /// If [fillColor] returns a non-null color in the [WidgetState.selected]
+  /// state, it will be used instead of this color.
+  ///
   /// Defaults to [CupertinoColors.activeBlue].
   final Color? activeColor;
 
+  /// {@template flutter.material.checkbox.fillColor}
+  /// The color used to fill this checkbox.
+  ///
+  /// Resolves in the following states:
+  ///  * [WidgetState.selected].
+  ///  * [WidgetState.hovered].
+  ///  * [WidgetState.focused].
+  ///  * [WidgetState.disabled].
+  ///
+  /// {@tool snippet}
+  /// This example resolves the [fillColor] based on the current [WidgetState]
+  /// of the [CupertinoCheckbox], providing a different [Color] when it is
+  /// [WidgetState.disabled].
+  ///
+  /// ```dart
+  /// CupertinoCheckbox(
+  ///   value: true,
+  ///   onChanged: (_){},
+  ///   fillColor: WidgetStateProperty.resolveWith<Color>((Set<WidgetState> states) {
+  ///     if (states.contains(WidgetState.disabled)) {
+  ///       return Colors.orange.withOpacity(.32);
+  ///     }
+  ///     return Colors.orange;
+  ///   })
+  /// )
+  /// ```
+  /// {@end-tool}
+  /// {@endtemplate}
+  ///
+  /// If null, then the value of [activeColor] is used in the selected
+  /// state. If that is also null, then [CupertinoColors.inactiveGray] is used in
+  /// the disabled state, and [CupertinoColors.activeBlue] is used in the
+  /// selected state.
+  final WidgetStateProperty<Color?>? fillColor;
+
   /// The color used if the checkbox is inactive.
   ///
+  /// If [fillColor] returns a non-null color in the [WidgetState.disabled]
```

Comment:
I made this mistake myself. For `fillColor` to be used, it must return a non-null color in unselected states (i.e states not containing `WidgetState.selected`). `inactiveColor` has nothing to do with `WidgetState.disabled`.